tests/events.cpp

```diff
@@ -959,18 +959,16 @@ TEST(ToDevice, KeyRequest)
         "sender": "@mujx:matrix.org",
         "type": "m.room_key_request"
 	})"_json;
-
-        ns::msg::KeyRequest event = request_data;
+        mtx::events::DeviceEvent<ns::msg::KeyRequest> event(request_data);
```

Comment:
Can you change the other ToDevice tests to also test agains mtx::events::DeviceEvent instead of ns::Event?

Reply:
The rest of them to_device events have only **content** and **type** fields hence only ns::Event has to be used.But we can change everything into DeviceEvent if we can make the sender type in the DeviceEvent optional.